### ISSUE TITLE
[Feature Request]: Alternative Install Method For Linux Distros

### ISSUE THREAD

**Issue by Majoraslayer** [lllyasviel/Fooocus#3534]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What would your feature do?

I've been using Fooocus on Windows for several months, but after trying to migrate to Linux Mint, I was surprised to find that all of the install methods for Linux use pip.  Every Linux distro seemingly ships with a different version and configuration of Python, and speaking from years of experience, always involves dependency hell to get working since Python is terrible with backwards compatibility.  Changing to different versions of Python to accommodate these issues causes all sorts of compatibility problems.  It would be great if Fooocus releases could include a .tgz package (maybe even a .deb and .rpm), or at the very least a cmake script that will compile everything without needing pip to download it.  In my specific case, Mint 22 ships with Python 3.12 and pip is configured to refuse installs with "error: externally-managed-environment".  If there's no way around this because of Fooocus relying on Python, maybe an official self-contained Docker image would be a good solution as well.

### Proposed workflow

1. Go to Releases page
2. Download .deb or .rpm package depending on Linux distro
3. Use the system's package installer (or extractor) to extract working binaries that Fooocus requires to run.

**Comment by zanaviska** [comment]
Maybe flathub app would be a good solution?

**Comment by Majoraslayer** [comment]
> Maybe flathub app would be a good solution?

A flatpak or snap could be a potential solution.  They aren't my preferred method because they have permissions issues with secondary storage drives, but it would be one way to make sure all dependencies are included with Fooocus.

**Comment by jsprenkle** [comment]
This is still a problem years later.

> pip3 install -r requirements_versions.txt
error: externally-managed-environment

Python version hell.

**Comment by Majoraslayer** [comment]
> This is still a problem years later.
> 
> > pip3 install -r requirements_versions.txt
> > error: externally-managed-environment
> 
> Python version hell.

Agreed.  An official Flatpak or Docker image would still be the easiest solution, but it looks like this project is mostly abandoned at this point.  I kind of abandoned using it myself when self-hosted AI models completely stagnated beyond usefulness, at least compared to what can be had for free from Grok or ChatGPT.  Here's hoping publicly available models finally start evolving again..
